### ISSUE TITLE
bprobe reports multiple stockpiles in the same tile

### ISSUE THREAD

**Issue by kelvie** [DFHack/dfhack#2621]
```
~ dfhack-run bprobe                                                                                                                      0 [21:48:13]
Building 1 - "Wood Stockpile #1" - type Stockpile (29)
Building 4 - "Activity Zone #1" - type Civzone (30), subtype Pen (88)
Building 431 - "Stockpile #11" - type Stockpile (29)

```

Of course this isn't actually possible
![image](https://user-images.githubusercontent.com/5893/212607890-7840ba5c-7ddf-4711-ad6d-7ea12c60e37a.png)

**Comment by kelvie** [comment]
If it helps, this is actually the Wood stockpile (building 1), and stockpile 11 (building 431) is sort of all around it.

**Comment by jcoskej** [comment]
It looks like it checks the if the cursor is in the rectangle created by the top left/bottom right coordinates, since stockpiles can not be all kinds of crazy shapes it looks like it'll need to check each tile of stockpiles instead of just the corners.

**Comment by ab9rf** [comment]
yeah, i checked the code, it only checks for being in the bounding box of the building, no extents checking is done

it's always been like this, too going back all the way to 2012. i think the idea is that the user is expected to interpret the output so it's not so much wrong as "failing to meet user expectations"
